compiler/rustc_resolve/src/ident.rs

```diff
@@ -1549,6 +1549,26 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                     ignore_import,
                 )
                 .map_err(|(determinacy, _)| determinacy)
+                .or_else(|err| {
+                    // HACK: try to resolve as a trait or type if it's the first segment in a `use` item.
```

Comment:
I'm afraid it is not the right way to do this. Are there any better solutions?

Reply:
I'd expext these to work similarly to `use LocalEnum::Variant`. Maybe check how that code path is implemented. Although it's in the value namespace, I'd expect some reasoning to be similar

Reply:
I added this hack because I failed to resolve `use Default::default`, but I finally found it is because of the edition 2015, as `use Option::Some` doesn't work either. Anyway, thanks for your hints!